### PR TITLE
Add plot_mask to RegionNDMap

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR should solve the issue #3271. 
It adds a `plot_mask` function into `RegionNDMap`, in order to take into account the correct region of the mask fit.

### CODE REVIEW

File: gammapy/maps/regionnd.py [hunk 1]
```diff
@@ -154,6 +155,37 @@ def plot_region(self, ax=None, **kwargs):
         ax = self.geom.plot_region(ax, **kwargs)
         return ax
 
+    def plot_mask(self, ax=None, color="k", label=""):
```
Comment (adonath) [inline gammapy/maps/regionnd.py:158]: I would propose to use the `**kwargs` pattern here, like we do for the other plot functions. Internally we can set default values for the `kwargs` like:

```python3

def plot_something(ax, **kwargs):
    """
    Parameters
    -----------
    **kwargs : dict
        Keyword arguments passed to `~matplotlib.pyplot.axvspan` 
    """
    kwargs.setdefault("color", "k")
    kwargs.setdefault("alpha", 0.05)
    kwargs.setdefault("label", "mask")
    ax.axvspan(xmin, xmax, **kwargs)


```

File: gammapy/maps/regionnd.py [hunk 2]
```diff
@@ -154,6 +155,37 @@ def plot_region(self, ax=None, **kwargs):
...
+        """Plot region
```
Comment (adonath) [inline gammapy/maps/regionnd.py:159]: Maybe improve the docstring here a bit to be more explicit on plotting the mask as shaded are etc.

File: gammapy/maps/regionnd.py [hunk 3]
```diff
@@ -154,6 +155,37 @@ def plot_region(self, ax=None, **kwargs):
...
+
+        Parameters
+        ----------
+        ax : `~astropy.vizualisation.WCSAxes`
+            Axes to plot on. If no axes are given,
+            the region is shown using the minimal
```
Comment (adonath) [inline gammapy/maps/regionnd.py:165]: This is a copy and paste error: here the ax is not a `WCSAxes` and there is no region plotting, but only a spectrum...

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @fabiopintore, I have left a few inline comments...

File: gammapy/maps/regionnd.py (hunk 1 above)
Comment (fabiopintore) [reply gammapy/maps/regionnd.py]: Done.

File: gammapy/maps/regionnd.py (hunk 2 above)
Comment (fabiopintore) [reply gammapy/maps/regionnd.py]: Done.

File: gammapy/maps/regionnd.py (hunk 3 above)
Comment (fabiopintore) [reply gammapy/maps/regionnd.py]: Ahh right, my fault! I've corrected it.

File: gammapy/maps/regionnd.py [hunk 4]
```diff
@@ -154,6 +155,37 @@ def plot_region(self, ax=None, **kwargs):
...
+    def plot_mask(self, ax=None, **kwargs):
+        """Plot the mask as a shaded area in a xmin-xmax range
+
+        Parameters
+        ----------
+        ax : `~matplotlib.axis` 
+            Axis instance to be used for the plot.
+        **kwargs : dict
+            Keyword arguments passed to `~matplotlib.pyplot.axvspan`
+        """
```
Comment (adonath) [inline gammapy/maps/regionnd.py:167]: Add `Returns` section here...you'll find example in the same file..

File: gammapy/maps/regionnd.py [hunk 5]
```diff
@@ -154,6 +155,37 @@ def plot_region(self, ax=None, **kwargs):
...
+        import matplotlib.pyplot as plt
+
```
Comment (adonath) [inline gammapy/maps/regionnd.py:169]: Include a check like `if not self.is_mask:` and raise an error to forbid the plotting of non-masks

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @fabiopintore, I have left two final comments..

File: gammapy/maps/regionnd.py (hunk 4 above)
Comment (fabiopintore) [reply gammapy/maps/regionnd.py:172]: Ok.

File: gammapy/maps/regionnd.py (hunk 5 above)
Comment (fabiopintore) [reply gammapy/maps/regionnd.py:174]: Ok.

**Review by adonath — APPROVED** [review]
Thanks @fabiopintore, no further comments from my side...

**Pull request merged**